packages/iov-bns/src/encode.ts

```diff
@@ -128,15 +130,10 @@ function encodeNumericId(id: number): Uint8Array {
 // Token sends
 
 function buildSendTransaction(tx: SendTransaction & WithCreator): codecImpl.bnsd.ITx {
-  const { prefix: prefix1, data: data1 } = decodeBnsAddress(identityToAddress(tx.creator));
-  const { prefix: prefix2, data: data2 } = decodeBnsAddress(tx.sender);
-  if (prefix1 !== prefix2 || data1.length !== data2.length || data1.some((b, i) => b !== data2[i])) {
-    throw new Error("Sender and creator do not match (currently unsupported)");
-  }
   return {
     cashSendMsg: codecImpl.cash.SendMsg.create({
       metadata: { schema: 1 },
-      source: decodeBnsAddress(identityToAddress(tx.creator)).data,
+      source: decodeBnsAddress(tx.sender).data,
```

Comment:
This change looks good to me. All those consistency checks were implemented when we added `sender` and did not have cases where sender would be different from creator.